x-pack/plugins/lens/public/editor_frame_plugin/editor_frame/expression_helpers.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Ast, fromExpression } from '@kbn/interpreter/common';
+import { Visualization, Datasource, DatasourcePublicAPI } from '../../types';
+
+export function buildExpression(
+  visualization: Visualization,
+  visualizationState: unknown,
+  datasource: Datasource,
+  datasourceState: unknown,
+  datasourcePublicAPI: DatasourcePublicAPI
+): Ast | undefined {
+  const datasourceExpression = datasource.toExpression(datasourceState);
+  const visualizationExpression = visualization.toExpression(
+    visualizationState,
+    datasourcePublicAPI
+  );
+
+  try {
+    const parsedDatasourceExpression =
+      typeof datasourceExpression === 'string'
+        ? fromExpression(datasourceExpression)
+        : datasourceExpression;
+    const parsedVisualizationExpression =
+      typeof visualizationExpression === 'string'
+        ? fromExpression(visualizationExpression)
+        : visualizationExpression;
+    return {
+      type: 'expression',
+      chain: [...parsedDatasourceExpression.chain, ...parsedVisualizationExpression.chain],
+    };
+  } catch (_) {
+    return undefined;
```

Comment:
Agreed, I moved the catch to the workspace panel and show the underlying error there.

<img width="854" alt="Screenshot 2019-06-04 at 10 05 45" src="https://user-images.githubusercontent.com/1508364/58862246-8aa5d280-86b0-11e9-85e6-0d5ad30f1b36.png">

I don't like letting the exception bubble up and crash Kibana because errors in the expression are a nice error boundary. In most cases it is probably possible to recover from the error without losing state.

We probably wont keep the very prominent error message in the workspace panel for the final release, but for beta it's nice to debug easily.